File: Common/InMotionGIT.Common - Copy/Helpers/RegistryHandler.vb
#Region "Imports"

Imports Microsoft

#End Region

Namespace Helpers

    ''' <summary>
    '''Class allows certified operations - Clase permite realizar operaciones sobre certificados
    ''' </summary>
    ''' <remarks></remarks>
    Public Class RegistryHandler

#Region "Private fields, to hold the state of the entity"

        Public Shared _regKey As Win32.RegistryKey

#End Region

#Region "Setting"

        Public Shared Sub SetSetting()
            _regKey = Win32.Registry.CurrentUser.OpenSubKey("Software\Global Insurance Technology", True)
            If IsNothing(_regKey) Then
                _regKey = Win32.Registry.CurrentUser.OpenSubKey("Software", True)
                CreateSubKey("Global Insurance Technology")

            End If
            release()
            _regKey = Win32.Registry.CurrentUser.OpenSubKey("Software\Global Insurance Technology\Settings", True)
            If IsNothing(_regKey) Then
                _regKey = Win32.Registry.CurrentUser.OpenSubKey("Software\Global Insurance Technology", True)
                CreateSubKey("Settings")
            End If
            release()
            _regKey = Win32.Registry.CurrentUser.OpenSubKey("Software\Global Insurance Technology\Settings", True)
        End Sub

#End Region

#Region "Main Methods"

        Public Shared Function GetValue(name As String, defaultValue As String) As String
            SetSetting()
            Return _regKey.GetValue(name, defaultValue)
        End Function

        Public Shared Sub SetValue(name As String, value As String)
            SetSetting()
            _regKey.SetValue(name, value, Win32.RegistryValueKind.String)
        End Sub

        Public Shared Sub release()
            _regKey.Close()
        End Sub

        Public Shared Sub CreateSubKey(ByVal name As String)
            SetSetting()
            _regKey.CreateSubKey(name)
        End Sub

#End Region

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/RegistryHandler.cs


#region Imports
using Microsoft.VisualBasic.CompilerServices;

#endregion

namespace InMotionGIT.Common.Helpers
{

    /// <summary>
    /// Class allows certified operations - Clase permite realizar operaciones sobre certificados
    /// </summary>
    /// <remarks></remarks>
    public class RegistryHandler
    {

        #region Private fields, to hold the state of the entity

        public static Microsoft.Win32.RegistryKey _regKey;

        #endregion

        #region Setting

        public static void SetSetting()
        {
            _regKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(@"Software\Global Insurance Technology", true);
            if (_regKey == null)
            {
                _regKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Software", true);
                CreateSubKey("Global Insurance Technology");

            }
            release();
            _regKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(@"Software\Global Insurance Technology\Settings", true);
            if (_regKey == null)
            {
                _regKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(@"Software\Global Insurance Technology", true);
                CreateSubKey("Settings");
            }
            release();
            _regKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(@"Software\Global Insurance Technology\Settings", true);
        }

        #endregion

        #region Main Methods

        public static string GetValue(string name, string defaultValue)
        {
            SetSetting();
            return Conversions.ToString(_regKey.GetValue(name, defaultValue));
        }

        public static void SetValue(string name, string value)
        {
            SetSetting();
            _regKey.SetValue(name, value, Microsoft.Win32.RegistryValueKind.String);
        }

        public static void release()
        {
            _regKey.Close();
        }

        public static void CreateSubKey(string name)
        {
            SetSetting();
            _regKey.CreateSubKey(name);
        }

        #endregion

    }

}